File: NodeJS/esercizio6.js
#!/usr/bin/env node

/*
Esercizio: scrivere una funzione che calcoli il fattoriale di un numero intero
non negativo
*/

'use strict';

function fattoriale(input) {
	if (input == 0 || input == 1) {
		return 1;
	}

	return input * fattoriale(input - 1);
}

var readline = require('readline-sync');

var input = parseInt(readline.question("Inserisci un numero intero (piccolo!): "));

// Calcola il fattoriale
var output = fattoriale(input);

// Stampo il risultato
console.log("Il fattoriale di " + input + " è " + output + ".");

File: C/esercizio6.c
/*
Esercizio: Scrivere la funzione fattoriale in modo ricorsivo, cioe facendo in modo che la funzione che calcola il fattoriale chiami se stessa.
*/

#include <stdio.h>
#include <stdlib.h>

int fattoriale(int input) {
	if (input == 0 || input == 1) {
		return 1;
	}

	return input * fattoriale(input - 1);
}

int main() {
	int input;
	
	// Richiedi il numero
	printf("Inserisci un numero intero (piccolo!): ");
	scanf("%d", &input);

	// Calcola il fattoriale
	int output = fattoriale(input);

	// Stampo il risultato
	printf("Il fattoriale di %d è %d.\n", input, output);

	return 0;
}

